packages/xml-views-completion/api.d.ts

```diff
@@ -122,3 +129,38 @@ export interface UI5NamespacesInXMLAttributeValueCompletion
   extends BaseXMLViewCompletion<XMLAttribute, UI5Namespace> {
   type: "UI5NamespacesInXMLAttributeValue";
 }
+
+export type LiteralXMLViewCompletionBaseType = boolean | string | number;
+export interface BaseLiteralXMLViewCompletion<
+  XML extends XMLAstNode,
+  Primitive extends PrimitiveType,
+  // This type must correspond to Primitive and have a string representation
+  LiteralXMLViewCompletionBaseType
+> {
+  type: LiteralXMLViewCompletionTypeName;
+  // The name of the suggestion
+  name: string;
+  // The primitive type of the we want to suggest as a possible completion.
+  primitiveType: Primitive;
+  // The literal value that we want to suggest as a possible completion.
+  value: LiteralXMLViewCompletionBaseType;
+
+  // The specific ASTNode where the completion happened
+  // may be useful for LSP Layer to implement Editor Level Logic.
+  //   - e.g: the "additional text insertions" mentioned above.
+  astNode: XML;
+}
+
+export type LiteralXMLViewCompletion = BooleanValueInXMLAttributeValueCompletion;
+
+export interface BooleanValueInXMLAttributeValueCompletion
+  extends BaseLiteralXMLViewCompletion<XMLAttribute, PrimitiveType, boolean> {
+  type: "BooleanValueInXMLAttributeValueCompletion";
+}
+
+export function isLiteralXMLViewCompletion(
```

Comment:
If we need to expose public API utils to distinguish the suggestions types
we may have too many types of suggestions.